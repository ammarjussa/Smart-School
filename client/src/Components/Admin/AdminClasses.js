import React, { Component } from 'react';
import {Route,Switch} from 'react-router'
import {Table, InputGroup, Row, Col, Nav } from 'react-bootstrap'
import { 
    Card,
    Spinner,
    OverlayTrigger,
    Tooltip,
    Button,
    Container,
    Breadcrumb,
    Modal,
    Form,
    Navbar
} from 'react-bootstrap';
import { FaEye,FaTrashAlt,FaPlus} from "react-icons/fa";
import { Input, ModalHeader } from 'semantic-ui-react';
import axios from "axios"


class ClassesTab extends Component {

    state = {
      classesData: [],
      classesDataBackup: [],
      selectAll: false,
      view_object: {},
      view_isview: false,
      add_showmodal:false,
      add_studentid: null,
      theclass: '',
      cap: '',
      section: '',
      delete_last_deleted: {},
      delete_last_deleted_id: null,
      delete_showmodal: false,
      adding: false,
      Search: '',
      showedit: false,
  }
  componentDidMount(){
      //Extract data 
      let resp;
      // axios.post("/students","ack").then((res) => {
          
          //     //Make all check false
          //       this.setState({
              //           classesData: res.data.students,
              //           classesDataBackup : res.data.students
              //       })
              // }).catch((e) => alert(e))
              
      axios.post("/showclass","ack").then((res) => {
          
          //Make all check false
            this.setState({
                classesData: res.data.classes,
                classesDataBackup : res.data.classes,
            })
      }).catch((e) => console.log("MyError:",e))
  }


  handleChangeCheckbox = (e) => {
    console.log(e)
  }

  handleDeleteModalClose = () =>{
    
    var prevState = Object.assign({},this.state)
    prevState.delete_showmodal = false
    this.setState(prevState)
  } 

  handleDeleteModalShow = () =>{
    var prevState = Object.assign({},this.state)
    prevState.delete_showmodal = true
    this.setState(prevState)
  } 

  handleViewModalClose = () =>{
    var prevState = Object.assign({},this.state)
    prevState.view_isview = false
    this.setState(prevState)
    this.setState({showedit: false})
  } 

  handleViewModalShow = (ev) =>{
    var prevState = Object.assign({},this.state)
    prevState.view_isview = true
    this.setState(prevState)
  } 

  handleDelete = (e) => {

    e.preventDefault()
    this.handleDeleteModalClose()
    // setTimeout(()=> this.handleDeleteModalClose(),3000)
    this.get_student(this.state.delete_last_deleted_id, (fObject) => {

        var prevState = Object.assign({},this.state)
        prevState.delete_last_deleted = fObject 
        this.setState(prevState, () => {
            let del_Obj = {
                id:fObject._id
            }

        axios.post("/deleteclass",del_Obj).then((res) => {
            // var prevState = Object.assign({},this.state)
            // prevState.classesData = this.state.classesData.filter((f) => f.studentid !== del_Obj.studentid && f.email !== del_Obj.email)
            // this.setState(prevState)
            console.log("Deleted", del_Obj.id)
            var prevState = Object.assign({},this.state)
            let fd = prevState.classesDataBackup
            fd = fd.filter((f) => f._id !== fObject._id)
            prevState.classesData = fd
            prevState.classesDataBackup = fd
            this.setState(prevState)
            this.handleDeleteModalClose()
        }).catch((e) => alert(e))

        })


    })
  }

  get_student(id,cb) {
    let count = 0
    this.state.classesData.forEach(f => {
        if(f._id===id && count<1)
        {
            count = count + 1
            console.log(f)
            cb(f)
        }
    });
  }

  handleDeleteTid = (e,id)  =>{
    var prevState = Object.assign({},this.state)
    prevState.delete_showmodal=true
    prevState.delete_last_deleted_id = id
    this.setState(prevState)
  }

  handleView = (e,id) => {
    this.get_student(id,(fObject) => {
        this.setState({
            view_object: fObject,
            view_isview : true
        })

    })
}



  handleAddModalClose = () =>{
    var prevState = Object.assign({},this.state)
    prevState.add_showmodal = false
    this.setState(prevState)
  }

  handleAddModalShow = () =>{
    var prevState = Object.assign({},this.state)
    prevState.add_showmodal = true
    this.setState(prevState)
  }

  handleSubmitAddStudent = (e) => {
    e.preventDefault()
    let obj = this.state
    if(obj.theclass==="" || obj.theclass==="Grade...")
    {
        alert("Please select a class")
    }
    else if(obj.section==="" || obj.theclass==="Section...")
    {
        alert("Please select a section")
    }
    else
    {
        this.setState({adding:true})
        let classInfo = {
            
            section: obj.section,
            theclass: obj.theclass,
            cap: obj.cap,
        }      
        console.log(classInfo)
        axios.post("/addclass",classInfo).then((res)=>{
            console.log(res.data.message)
            this.setState({adding:false})
            if(res.data.message === "Success")
            {
                this.setState({adding:false})
                let classInfo = {
                    _id: res.data.theid,
                    section: obj.section,
                    theclass: obj.theclass,
                    cap: obj.cap,
                } 

                var prevState = Object.assign({},this.state)
                prevState.add_showmodal = false
                prevState.theclass= ''
                prevState.cap= ''
                prevState.section= ''
                let fd = prevState.classesData
                fd.push(classInfo)
                prevState.classesData = fd
                prevState.classesDataBackup = fd
                this.setState(prevState)
            }
            else{
                alert("Something went Wrong!")
            }
            this.setState({add_showmodal:false})


        }).catch((e) => console.log(e))
    }
  }

  handleEdit = () => {
    this.setState({showedit: true})
  }


  handleEditUpdate = () => {
    this.setState({updating: true})
    let obj = this.state.view_object
        
       
    let classInfo = {
        
        id:obj._id,
        section: obj.section,
        theclass: obj.theclass,
        cap: obj.cap,
    }      

        console.log(classInfo)
        axios.post("/updateclass",classInfo).then((res)=>{
            console.log(res.data.message)
            if(res.data.message === "Success")
            {
                this.setState({updating:false})
                // var prevState = Object.assign({},this.state)
                // prevState.add_showmodal = false
                // prevState.view_object.theclass= ''
                // prevState.view_object.subject= ''
                // prevState.view_object.gender= ''
                // prevState.view_object.name= ''
                // prevState.view_object.contact= ''
                // prevState.view_object.email= ''
                // let fd = prevState.classesData
                // prevState.classesData = fd
                // this.setState(prevState)
                this.setState({showedit: false})
            }
            else{
                alert("Something went Wrong!")
            }


        }).catch((e) => console.log(e))
}

  handleAddInputs = (ev) => {
    ev.preventDefault()
    let {name,value} = ev.target
    console.log(name,value)
    var prevState = Object.assign({},this.state)
    prevState[name] = value
    this.setState(prevState)
  }

  handleEditInputs = (ev) => {
    ev.preventDefault()
    let {name,value} = ev.target
    console.log(name,value)
    var prevState = Object.assign({},this.state)
    prevState.view_object[name] = value
    this.setState(prevState)
}


adding = (e) =>{
    e.preventDefault()
    this.setState({adding: true})
}

  handleSearch = (e) => {
    e.preventDefault()
    this.setState({Search:e.target.value}, ()=>{
        console.log(this.state.Search)
        let tempvalue = this.state.Search
        if (tempvalue)
        {
            this.setState((prevState) => {
                    return {classesData: prevState.classesDataBackup.filter((f)=> f.theclass===tempvalue)}
            })
            
        }
        else
        {
            this.setState((prevState) => {
                return {classesData: prevState.classesDataBackup}
            })

        }
    })
    
    
  }


  render() {
    return (
      <div>
        <h1>Classes Information</h1>
        <div>
                {/* BreadCrumbs */}
                <Breadcrumb>
                    {/* <Breadcrumb.Item href="/admin/Dashboard">Admin</Breadcrumb.Item> */}
                    <Breadcrumb.Item>Admin</Breadcrumb.Item>
                    <Breadcrumb.Item active>View Classes</Breadcrumb.Item>
                </Breadcrumb>

                {/* Delete Modal */}
                <Modal
                    show={this.state.delete_showmodal}
                    onHide={this.handleDeleteModalClose}
                    size="sm"
                    centered
                >
                    <Modal.Header closeButton>
                        <Modal.Title>Are you sure?</Modal.Title>
                    </Modal.Header>
                    <Modal.Footer>
                        <Button variant="secondary" onClick={this.handleDeleteModalClose}>No</Button>
                        <Button variant="danger" onClick={this.handleDelete}>Yes</Button>
                    </Modal.Footer>
                </Modal>


                {/* View/Edit Model */}
                <Modal
                    show={this.state.view_isview}
                    onHide={this.handleViewModalClose}
                    size="sm"
                    centered
                >
                    <Modal.Header closeButton>
                    {this.state.showedit?
                        <Modal.Title>Edit class Information</Modal.Title>
                        :
                        <Modal.Title>class Information</Modal.Title>}
                    </Modal.Header>
                    <Modal.Body>
                        {this.state.showedit?
                              <Form onSubmit={this.handleSubmitAddStudent}>
                              <div>

                                  <Form.Row>

                                      <Form.Group as={Col}>
                                      <Form.Label>class:</Form.Label>
                                      <Form.Control 
                                       name="theclass"
                                       value={this.state.view_object.theclass}
                                       disabled
                                       onChange = {this.handleEditInputs}
                                      as="select">                                             <option>Grade...</option>
                                          <option>5</option>
                                          <option>6</option>
                                          <option>7</option>
                                          <option>8</option>
                                          <option>9</option>
                                      </Form.Control>
                                      </Form.Group>
                                  </Form.Row>
                                  <Form.Row>

                                      <Form.Group as={Col}>
                                      <Form.Label>Section:</Form.Label>
                                      <Form.Control 
                                       name="section"
                                       disabled
                                       value={this.state.view_object.section}
                                       onChange={this.handleEditInputs}
                                      as="select">
                                          <option>Section...</option>
                                          <option>A</option>
                                          <option>B</option>
                                          <option>C</option>
                                          <option>D</option>
                                          <option>E</option>
                                          <option>F</option>
                                          <option>G</option>
                                          <option>H</option>
                                          <option>I</option>
                                          <option>J</option>
                                          <option>K</option>
                                          <option>L</option>
                                      </Form.Control>
                                      </Form.Group>
                                  </Form.Row>

                                  <Form.Row>
                                  <Form.Group as={Col}>
                                  <Form.Label>Cap:</Form.Label>
                                      <Form.Control
                                          required
                                          type="number"
                                          min = "20"
                                          max = "50"
                                          placeholder="Enter class Cap"
                                          name="cap"
                                          value={this.state.view_object.cap}
                                          onChange={this.handleEditInputs}
                                          />
                                    {/* <Form.Text className="text-muted">
                                        {"20 <= class Size <= 50"} 
                                    </Form.Text> */}
                                  </Form.Group>
                                     
                                  </Form.Row>

                                  </div>
                          </Form>
                        
                        
                        :
                        <Container>
                            <Row>
                                <Col>
                                    <b>class:</b>
                                </Col>
                                <Col>
                                    {this.state.view_object.theclass}
                                </Col>
                            </Row>
                            <Row>
                                <Col>
                                    <b>Section:</b>
                                </Col>
                                <Col>
                                    {this.state.view_object.section}
                                </Col>
                            </Row>
                            <Row>
                                <Col>
                                    <b>Cap:</b>
                                </Col>
                                <Col>
                                    {this.state.view_object.cap}
                                </Col>
                            </Row>
                        </Container>}
                    </Modal.Body>
                    <Modal.Footer>
                    {this.state.showedit?
                        <div>
                            {
                            this.state.updating?
                                <Button variant="warning" disabled >Updating...</Button>
                                :
                                <Button variant="warning" onClick={this.handleEditUpdate}>Update</Button>
                            }
                        </div>
                        :
                        <div>
                        {/* <Button variant="secondary" onClick={this.handleViewModalClose}>Close</Button> */}
                        <Button variant="danger" onClick={this.handleEdit}>Edit</Button>
                        </div>}
                    </Modal.Footer>
                </Modal>

                {/* Add Faculty */}
                <Modal
                    show={this.state.add_showmodal}
                    onHide = {()=>this.setState({add_showmodal: false})}
                    size="sm"
                    centered
                >
                    <Modal.Header > 
                        <Modal.Title>Add class</Modal.Title>
                    </Modal.Header>
                        <Form onSubmit={this.handleSubmitAddStudent}>
                            <div>
                    <Modal.Body>
                        {/* <Card>
                            <Ca rd.Body>*/}

                                        <Form.Row>

                                            <Form.Group as={Col}>
                                            <Form.Label>class:</Form.Label>
                                            <Form.Control 
                                             name="theclass"
                                             value={this.state.theclass}
                                             onChange = {this.handleAddInputs}
                                            as="select">                                             <option>Grade...</option>
                                                <option>5</option>
                                                <option>6</option>
                                                <option>7</option>
                                                <option>8</option>
                                                <option>9</option>
                                            </Form.Control>
                                            </Form.Group>
                                        </Form.Row>
                                        <Form.Row>

                                            <Form.Group as={Col}>
                                            <Form.Label>Section:</Form.Label>
                                            <Form.Control 
                                             name="section"
                                             value={this.state.section}
                                             onChange={this.handleAddInputs}
                                            as="select">
                                                <option>Section...</option>
                                                <option>A</option>
                                                <option>B</option>
                                                <option>C</option>
                                                <option>D</option>
                                                <option>E</option>
                                                <option>F</option>
                                                <option>G</option>
                                                <option>H</option>
                                                <option>I</option>
                                                <option>J</option>
                                                <option>K</option>
                                                <option>L</option>
                                            </Form.Control>
                                            </Form.Group>
                                        </Form.Row>

                                        <Form.Row>
                                        <Form.Group as={Col}>
                                        <Form.Label>Cap:</Form.Label>
                                            <Form.Control
                                                required
                                                type="number"
                                                min = "20"
                                                max = "50"
                                                placeholder="Enter class Cap"
                                                name="cap"
                                                value={this.state.cap}
                                                onChange={this.handleAddInputs}
                                                />
                                        </Form.Group>
                                           
                                        </Form.Row>

                            {/* </Card.Body>
                        </Card>  */}
                    </Modal.Body>
                    <Modal.Footer>
                    {this.state.adding?
                            <Button disabled ><FaPlus/> Adding... </Button>
                            :
                            <Button type="submit" ><FaPlus/> Add </Button>
                       }
                    </Modal.Footer>
                        </div>
                </Form>
                </Modal>


                <br/>


                {/* Search Box and Add Faculty */}
                
                {/* <Navbar sticky="top">
                <Input type="text" value={this.state.Search} onChange={this.handleSearch} placeholder="Search Student name"></Input>
                      <Navbar.Collapse className="justify-content-end" >
                            <Button onClick={this.handleAddModalShow}> <FaPlus/> Add Student</Button>
                      </Navbar.Collapse>
                </Navbar> */}

                <Container>
                  <Row>
                    <Col xs={7} sm={9}>
                    <Input type="text" value={this.state.Search} onChange={this.handleSearch} placeholder="Search class"></Input>
                    </Col>
                    <Col>
                    <Button onClick={this.handleAddModalShow}> <FaPlus/> Add class</Button>
                    </Col>
                  </Row>
                </Container>



                {/* <div style={{flexDirection: 'row', justifyContent: 'space-between'}}>
                     <span style={{ paddingLeft: "10px" }}><Input type="text" value={this.state.Search} onChange={this.handleSearch} placeholder="Search Student name"></Input></span>
                     <span  style={{ paddingLeft: "600px" }}><Button onClick={this.handleAddModalShow}> <FaPlus/> Add Student</Button></span>
                </div>    */}

                <br/>

                {/* Table */}
                <Table responsive bordered striped hover>
                    <thead>
                        <th> id </th>
                        <th> class-Section</th>
                        <th>  Cap   </th>
                        <th>   </th>
                    </thead>
                    <tbody>

                        {
                            this.state.classesData.map(({section,theclass,cap,_id},id) => 
                                <tr key={id}>
                                    <td>{_id}</td>
                                    <td>{theclass}-{section}</td>
                                    <td>{cap}</td>
                                    <td>
                                        {/* <OverlayTrigger overlay={<Tooltip id="tooltip-disabled">View/Edit</Tooltip>}>
                                            <span className="d-inline-block"> */}
                                                 <Container>
                                                <Row>
                                                <Col xs={4.5}>
                                                
                                                <Button variant="success" onClick={(e)=>this.handleView(e,_id)}> <FaEye/> View/Edit</Button>
                                                </Col>
                                                <Col>
                                                <Button variant="danger" onClick={(e) => this.handleDeleteTid(e,_id)}> <FaTrashAlt/> Delete</Button>
                                                </Col>
                                                </Row>

                                            </Container>
                                            {/* </span>
                                            </OverlayTrigger> */}
                                    </td>
                                    
                                </tr>
                            )

                        }

                    </tbody>
                </Table>

            </div>
      </div>
    );
  }
}

export default ClassesTab;