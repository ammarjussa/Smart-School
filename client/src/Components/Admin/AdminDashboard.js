import React, { Component } from 'react';
import {Route,Switch} from 'react-router'
import auth from '../../auth';
import Profile from '../../Profile'

class AdminDashboard extends Component {
  render() {
    return (
      <div>
        <h1>Admin Dashboard</h1>
        <p> Welcome {Profile.getEmail()} </p>
      </div>
    );
  }
}

export default AdminDashboard;