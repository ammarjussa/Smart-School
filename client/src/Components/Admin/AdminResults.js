import React, { Component } from 'react';
import { Route, Switch } from 'react-router'
import auth from '../../auth';
import {Table, InputGroup, Row, Col } from 'react-bootstrap'
import { 
    Card,
    Spinner,
    OverlayTrigger,
    Tooltip,
    Button,
    Container,
    Breadcrumb,
    Modal,
    Form
} from 'react-bootstrap';
import { FaEye,FaTrashAlt,FaPlus,FaArrowRight, FaCheck} from "react-icons/fa";
import { Input, ModalHeader } from 'semantic-ui-react';
import axios from "axios"


const renderTooltip = props => (
  <div
    {...props}
    style={{
      backgroundColor: 'rgba(0, 0, 0, 0.85)',
      padding: '2px 10px',
      color: 'white',
      borderRadius: 3,
      ...props.style,
    }}
  >
    Send Again? Click Here!
  </div>
);


class ResultsTab extends Component {

  state = {
    studentData: [],
    studentDataBackup: [],
    selectAll: false,
    view_isview: false,
    view_object: {},
    edit:{
        id: '',
        isEdit: false
    },
    add_showmodal:false,
    add_studentid: null,
    add_class: '',
    add_section: '',
    add_gender: '',
    add_name: '',
    add_pname: '',
    add_pcontact: '',
    add_pemail: '',
    delete_last_deleted: {},
    delete_last_deleted_id: null,
    delete_showmodal: false,
    adding: false,
    Search: ''
}


componentDidMount(){
    //Extract data 
    let resp;
    // axios.post("/students","ack").then((res) => {
        
        //     //Make all check false
        //       this.setState({
            //           studentData: res.data.students,
            //           studentDataBackup : res.data.students
            //       })
            // }).catch((e) => alert(e))
            
    let path = "/students"
    axios.post("/students","ack").then((res) => {
        
        //Make all check false
          this.setState({
              studentData: res.data.students,
              studentDataBackup : res.data.students
          })
    }).catch((e) => alert(e))
}


updateIsSent = (students,studentsbacku,id) => {
    students.forEach((student) => {
        if(student._id===id)
        {
          student.issent=true
          this.setState({
            studentData: students,
            studentDataBackup: students
          })
        }
    })
}


handleSendResult(e,id,name) {
  console.log(id)
  axios.post("/send-email",{id}).then((res) => {
        
    // Make all check false
     this.updateIsSent(this.state.studentData,this.state.studentDataBackup,id)
}).catch((e) => console.log(e))
}



  render() {
    return (
      <div>
        <h1>Results Information</h1>
        <div>
                {/* BreadCrumbs */}
                <Breadcrumb>
                    {/* <Breadcrumb.Item href="/admin/Dashboard">Admin</Breadcrumb.Item> */}
                    <Breadcrumb.Item>Admin</Breadcrumb.Item>
                    <Breadcrumb.Item active>Send Results</Breadcrumb.Item>
                </Breadcrumb>

                <br/>
  
                {/* Table */}
                <Table responsive bordered striped hover>
                    <thead>
                        {/* <th><InputGroup.Checkbox checked={this.state.selectAll} onChange={this.handleChangeSelectAll}/></th> */}
                        <th> id</th>
                        <th> Name</th>
                        <th> Class-Section </th>
                        <th> Email </th>
                        <th>     </th>
                    </thead>
                    <tbody>

                        {
                            this.state.studentData.map(({section,theclass,name,pemail,_id,issent},id) => 
                                <tr key={id}>
                                    {/* <td><InputGroup.Prepend><InputGroup.Checkbox /> </InputGroup.Prepend> </td> */}
                                    <td>{_id}</td>
                                    <td>{name}</td>
                                    <td>{`${theclass}-${section}`}</td>
                                    <td>{pemail}</td>
                                    <td>
                                      {
                                        issent?
                                        <OverlayTrigger
                                        placement="left-end"
                                        delay={{show:250, hide:100}}
                                        overlay={renderTooltip}
                                        >

                                          <Button variant="success" onClick={(e) => this.handleSendResult(e,_id,name)}> <FaCheck/> Sent  </Button>
                                        </OverlayTrigger>


                                        :
                                    
                                        <Button variant="secondary" onClick={(e) => this.handleSendResult(e,_id,name)}>  Send <FaArrowRight/> </Button>
                                    }
                                    </td>
                                    
                                </tr>
                            )

                        }

                    </tbody>
                </Table>

        </div>
      </div>
    );
  }
}

export default ResultsTab;