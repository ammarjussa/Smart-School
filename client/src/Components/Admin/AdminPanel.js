import React, { Component } from 'react';
import {Route,Switch} from 'react-router'
import auth from '../../auth';
import { Header, Icon, Image, Menu, Segment, Sidebar } from 'semantic-ui-react'
import AdminDashboard from './AdminDashboard'
import NotFound from '../NotFound'
import Faculty from './AdminFaculty'
import {FormControl,Form,Navbar,Tab,Nav,Table, Alert, Modal, Container, Col, Row, Card, Button } from 'react-bootstrap';
import { FaPowerOff,FaTachometerAlt,FaUserGraduate,FaUserFriends,FaMailBulk,FaLandmark} from "react-icons/fa";
import ResultsTab from './AdminResults';
import StudentsTab from './AdminStudents';
import ClassesTab from './AdminClasses';

class AdminPanel extends Component {
  
  handleItemClick = (eventKey) => {
    this.props.history.push("/admin/" + eventKey)
  }

  componentDidMount(){
    this.props.history.push("/admin/Dashboard")
  }

  logout = () => {
    auth.logout(()=> {
      this.props.history.push("/")
    })
  }

  componentWillUnmount(){
    auth.login(()=> {
      this.props.history.push("/")
    })
  }

  render() {
    return (
      <div>
        <Navbar style={{backgroundColor:"grey"}} sticky="top">
          <Navbar.Brand style={{fontSize: "20px"}}>Smart School Managment System</Navbar.Brand>
          <Navbar.Collapse className="justify-content-end" >
          <Button variant="outline-warning" onClick={this.logout} > <FaPowerOff/> Logout</Button>
          
          </Navbar.Collapse>
        </Navbar>

        <Tab.Container id="left-tabs-example" defaultActiveKey="Dashboard">
          <Row>
            <Col sm={2}>
              <Nav sticky="top" variant="pills" className="flex-column">
                <Nav.Item >
                  <Nav.Link style={{content:"50px",height:"50px"}} onSelect={this.handleItemClick} eventKey="Dashboard"><FaTachometerAlt /> Dashboard</Nav.Link>
                </Nav.Item>
                <Nav.Item>
                  <Nav.Link style={{content:"50px",height:"50px"}} onSelect={this.handleItemClick} eventKey="Students"><FaUserFriends/> Students</Nav.Link>
                </Nav.Item>
                <Nav.Item>
                  <Nav.Link style={{content:"50px",height:"50px"}} onSelect={this.handleItemClick} eventKey="Classes"><FaLandmark/> Classes</Nav.Link>
                </Nav.Item>
                <Nav.Item>
                  <Nav.Link style={{content:"50px",height:"50px"}} onSelect={this.handleItemClick} eventKey="Faculty"><FaUserGraduate/> Faculty</Nav.Link>
                </Nav.Item>
                <Nav.Item>
                  <Nav.Link style={{content:"50px",height:"50px"}} onSelect={this.handleItemClick} eventKey="Results"><FaMailBulk/> Send Results</Nav.Link>
                </Nav.Item>
              </Nav>
            </Col>
            <Col sm={9}>
              <Tab.Content>
                {/* <Tab.Pane eventKey="Dashboard">            
                      <AdminDashboard/>
                </Tab.Pane>
                <Tab.Pane eventKey="Students">            
                      <NotFound/>
                </Tab.Pane>
                <Tab.Pane eventKey="Classes">            
                      <NotFound/>
                </Tab.Pane>
                <Tab.Pane eventKey="Faculty"> 
                      {}           
                      <Faculty/>
                </Tab.Pane>
                <Tab.Pane eventKey="Results">            
                      <NotFound/>
                </Tab.Pane> */}
                <br/>
                <Switch>
                <Route path="/admin/Dashboard" component={AdminDashboard}/>
                <Route path="/admin/Students" component={StudentsTab}/>
                <Route path="/admin/Classes" component={ClassesTab}/>
                <Route path="/admin/Faculty" component={Faculty}/>
                <Route path="/admin/Results" component={ResultsTab}/>
                <Route path="/admin/*" component={NotFound}/>
              </Switch>
              </Tab.Content>
            </Col>
          </Row>
        </Tab.Container>
      </div>
    );
  }
}

export default AdminPanel;
