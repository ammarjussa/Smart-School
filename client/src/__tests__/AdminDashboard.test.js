import { AdminDashboard } from '../Components/Admin/AdminDashboard';
import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';

const component = shallow(<AdminDashboard />);

describe('Admin Dashboard Component', () => {
    it('Admin Dashboard renders successfully', () => {
        // const wrapper = shallow(<AdminDashboard />)
        expect(toJson(component)).toMatchSnapshot()
    }) ; 

    it('Renders a H1 tag' , () => {
        const wrapper= component.find('h1')
        expect(wrapper.length).toBe(1)
    }) ; 
    it('Renders a P tag', () => {
        const wrapper = component.find('p')
        expect(wrapper.length).toBe(1)
    }); 
});