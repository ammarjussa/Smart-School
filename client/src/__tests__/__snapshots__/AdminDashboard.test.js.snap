// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Dashboard Component Admin Dashboard renders successfully 1`] = `
<div>
  <h1>
    Admin Dashboard
  </h1>
  <p>
     Welcome 
     
  </p>
</div>
`;
